From 74ccdfbf95356e7c84631f09aaf5e25ea0867274 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Date: Wed, 4 Dec 2024 23:26:12 +0100
Subject: [PATCH 1409/1695] dt-bindings: stm32_rng: add RNG customization
 optional properties

Add optional properties to set RNG configurations (NIST and health test)
for customization.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f121d45bb06a83f84b4721b9e58872944b983ebb dt-bindings: stm32_rng: add RNG customization optional properties]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Change-Id: Ic9f28845321f8221a2b7c807554333905f14ec93
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/422915
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
---
 .../devicetree/bindings/rng/st,stm32-rng.yaml | 30 +++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/documentation/devicetree/bindings/rng/st,stm32-rng.yaml b/documentation/devicetree/bindings/rng/st,stm32-rng.yaml
index 8711107f9..9f303aec4 100644
--- a/documentation/devicetree/bindings/rng/st,stm32-rng.yaml
+++ b/documentation/devicetree/bindings/rng/st,stm32-rng.yaml
@@ -73,6 +73,36 @@ allOf:
 
       required:
         - clock-names
+
+  - if:
+      properties:
+        compatible:
+          contains:
+            enum:
+              - st,stm32mp13-rng
+              - st,stm32mp21-rng
+              - st,stm32mp25-rng
+    then:
+      properties:
+        st,rng-cfg:
+          minItems: 1
+          maxItems: 3
+          type: boolean
+          description: The configuration bitfields to initialize the RNG. They
+                       must be listed in the numerical order. If not specified,
+                       the default hard-coded values will be used. Number of
+                       values depends on the presence of
+                       st,rng-nist-custom property.
+
+        st,rng-cfg-nist-custom:
+          type: boolean
+          description: Enable custom values for NIST compliant RNG.
+
+        st,rng-htcfg:
+          maxItems: 1
+          description: The values to configure health tests. If not specified,
+                       the default hard-coded values will be used.
+
 additionalProperties: false
 
 examples:
-- 
2.47.2

